agent/engine/docker_task_engine.go

```diff
@@ -346,8 +348,9 @@ func (engine *DockerTaskEngine) synchronizeContainerStatus(container *api.Docker
 		}
 		return
 	}
-
-	currentState, metadata := engine.client.DescribeContainer(container.DockerID)
+	ctx, cancel := context.WithCancel(engine.ctx)
```

Comment:
why are you doing this? Can't you just use `engine.ctx` directly?

Reply:
Will change them all to engine.ctx.